pkg/master/pod_cache.go

```diff
@@ -221,6 +227,10 @@ func (p *PodCache) GarbageCollectPodStatus() {
 // calling again, or risk having new info getting clobbered by delayed
 // old info.
 func (p *PodCache) UpdateAllContainers() {
+	// TODO: this is silly, we should pro-actively update the pod status when
+	// the API server makes changes.
+	p.clearNodeStatus()
```

Comment:
It is periodically updated, given that we clear the cache, and then update pod status, which forces an update to the nodeStatus too.  If you'd prefer to have two different go routines doing the periodic updates, that's ok with me too, although we're still going to want to have the read-through behavior as well to make sure that we have good data when a pod is initially created.

We could do it that way if we prefer.  That's not the way that it was previously structured.